ddtrace/appsec/_iast/reporter.py

```diff
@@ -217,7 +211,13 @@ def _to_dict(self) -> Dict[str, Any]:
         Returns:
         - Dict[str, Any]: Dictionary representation of the IAST span reporter.
         """
-        return attr.asdict(self, filter=lambda attr, x: x is not None and attr.name != "_ranges")
+        return attr.asdict(
+            self,
+            filter=lambda attr, x: x is not None
+            and attr.name != "_ranges"
+            and attr.name != "_evidences_with_no_sources"
+            and attr.name != "dialect",
```

Comment:
```suggestion
            and attr.name not in ("_ranges", "_evidences_with_no_sources", "dialect"),
```

Should be the same, right?

If you want to improve performance you could make it a `frozenset` constant.